### PR TITLE
feat(components): [el-time-select] support 12-hour format

### LINKED ISSUE #4488: [Feature Request] el-time-select Timepicker should support displaying in 12 hour format
### Existing Component
Yes

### Component Name
el-time-select

### Description
el-time-select should accept a prop to toggle the display label to use 12h or 24h format. Currently it's default to 24h format and can't be customized

### PR DESCRIPTION
**Pull request by Alanscut**
fix #4488 
support 12-hour format. (default use 24-hour format)
edit some description in docs, 

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/time-select/src/time-select.vue [hunk 1]
```diff
@@ -86,6 +91,21 @@ const nextTime = (time: string, step: string): string => {
...
+  const time = parseTime(current)
+  let hours = time.hours
+  if (hours === 0 || hours === 24) {
+    hours = 12
+  }
+  if (hours > 12) {
+    hours = hours - 12
+  }
+  const apm = time.hours >= 12 && time.hours !== 24 ? ' PM' : ' AM'
+  return `${hours < 10 ? `0${hours}` : hours}:${
```
Comment (btea) [inline packages/components/time-select/src/time-select.vue:104]: Maybe can use `padStart` 
```js
${(hours + '').padStart(2, '0')}
```

Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue:104]: ok,

File: packages/components/time-select/src/time-select.vue [hunk 2]
```diff
@@ -94,6 +116,10 @@ export default defineComponent({
     event: 'change',
   },
   props: {
+    use12h: {
+      type: Boolean,
```
Comment (btea) [inline packages/components/time-select/src/time-select.vue:120]: Maybe it's more appropriate to use this `use12Hours`

Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue:120]: hi @btea, you could insert a suggestion<ctrl + g> directly next time :)

Comment (btea) [reply packages/components/time-select/src/time-select.vue:120]: Okay, thanks for reminding, I didn’t know this usage before 😂

File: packages/components/time-select/src/time-select.vue [hunk 3]
```diff
@@ -48,8 +48,13 @@ interface Time {
 const parseTime = (time: string): null | Time => {
   const values = (time || '').split(':')
   if (values.length >= 2) {
-    const hours = parseInt(values[0], 10)
+    let hours = parseInt(values[0], 10)
     const minutes = parseInt(values[1], 10)
+    if (time.includes('AM') && hours === 12) {
+      hours = 0
+    } else if (time.includes('PM')) {
+      hours += 12
```
Comment (btea) [inline packages/components/time-select/src/time-select.vue:56]:
```suggestion
  if (hours !== 12) {
    hours += 12
  }
```

File: packages/components/time-select/src/time-select.vue (hunk 2 above)
Comment (msidolphin) [reply packages/components/time-select/src/time-select.vue:120]: I think we can judge by format, avoid add too many attributes

File: packages/components/time-select/src/time-select.vue (hunk 3 above)
Comment (btea) [inline packages/components/time-select/src/time-select.vue]:
```suggestion
      if (hours !== 12) {
        hours += 12
      }
```

Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue]: @btea please `pnpm lint:fix` before you commit and push your code

Comment (btea) [reply packages/components/time-select/src/time-select.vue]: Sorry, this is a misoperation, I will pay attention to it next time

File: packages/components/time-select/src/time-select.vue (hunk 2 above)
Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue:120]: > 

in my opinion, the time-select is just provide hours and minutes, i don't think we need more format. if users want to get more information or use more format they can use `time-picker`

Comment (msidolphin) [reply packages/components/time-select/src/time-select.vue:120]: > 

I'm sorry, I thought it was time-picker

**Review by btea — APPROVED** [review]

**Comment by ziw** [comment]
thanks a lot for fixing this!

**Comment by Alanscut** [comment]
@msidolphin 
cc

**Comment by Alanscut** [comment]
@msidolphin @JeremyWuuuuu

File: docs/en-US/component/time-select.md [hunk 4]
```diff
@@ -45,6 +47,7 @@ time-select/time-range
 | step                  | time step                                                | string             | —                       | 00:30       |
 | min-time              | minimum time, any time before this time will be disabled | string             | —                       | 00:00       |
 | max-time              | maximum time, any time after this time will be disabled  | string             | —                       | —           |
+| use-12-hour           | use 12-hour format                                       | boolean            | —                       | false       |
```
Comment (JeremyWuuuuu) [inline docs/en-US/component/time-select.md:50]: I'd suggest here we use `clock-format` for this feature:

```ts
type ClockFormat = '12h' | '24h'

const clockFormat = {
  type: ColckFormat,
  default: '24h',
  acceptanceValue: ['12h', '24h']
}
```

File: packages/components/time-select/src/time-select.vue [hunk 5]
```diff
@@ -48,8 +48,13 @@ interface Time {
 const parseTime = (time: string): null | Time => {
   const values = (time || '').split(':')
   if (values.length >= 2) {
-    const hours = parseInt(values[0], 10)
+    let hours = parseInt(values[0], 10)
     const minutes = parseInt(values[1], 10)
+    if (time.includes('AM') && hours === 12) {
+      hours = 0
+    } else if (time.includes('PM') && hours !== 12) {
+      hours += 12
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.vue:56]: **Suggestion:**  Can we do a safer check here?

e.g.

```ts
time = time.toLowerCase() // or toUpperCase()

if (time.includes('am') && hours = 12) { // or includes('AM')
}
```

File: docs/en-US/component/time-select.md (hunk 4 above)
Comment (Alanscut) [reply docs/en-US/component/time-select.md:50]: ok

File: packages/components/time-select/src/time-select.vue [hunk 6]
```diff
@@ -68,9 +73,10 @@ const compareTime = (time1: string, time2: string): number => {
   return minutes1 > minutes2 ? 1 : -1
 }
 const formatTime = (time: Time): string => {
-  return `${time.hours < 10 ? `0${time.hours}` : time.hours}:${
-    time.minutes < 10 ? `0${time.minutes}` : time.minutes
-  }`
+  return `${`${time.hours}`.padStart(2, '0')}:${`${time.minutes}`.padStart(
```
Comment (JeremyWuuuuu) [inline packages/components/time-select/src/time-select.vue:76]: For using this method about 4 times in this same file, you should consider use a function to encapsulate this:

```ts
function padTime(time: number | string) {
  return `${time}`.padStart(2, 0)
}
```

Then you use it like:

```ts
return `${padTime(time.hours)}:${padTime(time.minutes)}`
```

**Review by jw-foss — COMMENTED** [review]
I left few comments for you to consider, let me know if you have any questions. :bow:

File: packages/components/time-select/src/time-select.vue (hunk 5 above)
Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue:56]: ok

File: packages/components/time-select/src/time-select.vue (hunk 6 above)
Comment (Alanscut) [reply packages/components/time-select/src/time-select.vue:76]: ok

**Comment by emojiiii** [comment]
can you test ? I think the first option is error @Alanscut 
```vue
<el-time-select
    v-model="value"
    start="12:00 AM"
    step="00:30"
    end="23:59"
    placeholder="Select time"
  >
  </el-time-select>
```

**Comment by Alanscut** [comment]
> can you test case below ? I think the first option is error @Alanscut
> 
> ```
> <el-time-select
>     v-model="value"
>     start="12:00 AM"
>     step="00:30"
>     end="23:59"
>     placeholder="Select time"
>   >
>   </el-time-select>
> ```

you are right, it should display 00:00, but 12:00 now

**Comment by Alanscut** [comment]
@emojiiii 
it is ok now.

**Pull request closed without merging**